src/include/hardware.h

```diff
@@ -25,6 +25,16 @@
         WT_READ_BARRIER();      \
     } while (0)
 
+/*
+ * Read a shared location and guarantee that the value is only read once and subsequent reads do not
+ * see any earlier state in architectures with weak memory ordering.
+ */
+#define WT_ORDERED_READ_WEAK(v, val) \
```

Comment:
`WEAK` doesn't feel like the right name for this. I think you chose it as in: "for platforms with weak memory ordering". In reading it without context, it sounds like it's a weaker version of a barrier.

Could you find a name that is more definitive, even if it is verbose?

Reply:
How about `WT_ORDERED_READ_WEAK_MEMORDER`? I think the current name is a bit too verbose.

Reply:
I like that name better - it's a pain, but @quchenhao would you mind making the change to see how it looks?

Reply:
done